samcli/commands/local/start_api/cli.py

```diff
@@ -226,7 +240,13 @@ def do_cli(  # pylint: disable=R0914
                     ]
                 )
                 click.secho(command_suggestions, fg="yellow")
-
+    except SSLError as ex:
+        error_message = (
+            "SSL key file must be present if certificate is present"
+            if ssl_key_file is None
+            else "Invalid certificate and/or key file"
+        )
```

Comment:
My bad, I think I've read this line wrong at first, but now it makes more sense.

If you think `SSLError` text is good enough, we can bubble up the error or just its message back to the user. And once we are surfacing these errors, we need to make sure we are raising `UserError` so that CLI will handle them gracefully rather than crashing with exception trace (I can check that one later once you made the changes).

Reply:
I think the `SSLError` text is good enough (and I'm not sure we can cover all possible SSL Errors with our own text anyway), but I'll still catch it and wrap it in a `UserError`. The click options validator will catch the other case of mutually inclusive options.